docs/source/reference/kubernetes/kubernetes-deployment.rst

```diff
@@ -0,0 +1,235 @@
+4. After you are done using the cluster, you can remove it with :code:`sky local down`. This will terminate the KinD container and switch your kubeconfig back to it's original context:
+
+   .. code-block:: console
+
+     $ sky local down
+
+.. note::
+    We recommend allocating at least 4 or more CPUs to your docker runtime to
+    ensure kind has enough resources. See instructions
+    `here <https://docs.docker.com/desktop/settings/linux/>`_.
+
+.. note::
+    kind does not support multiple nodes and GPUs.
+    It is not recommended for use in a production environment.
+    If you want to run a private on-prem cluster, see the section on :ref:`on-prem deployment <kubernetes-setup-onprem>` for more.
+
+
+.. _kubernetes-setup-gke:
+
+Deploying on Google Cloud GKE
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Create a GKE standard cluster with at least 1 node. We recommend creating nodes with at least 4 vCPUs.
+2. Get the kubeconfig for your cluster. The following command will automatically update ``~/.kube/config`` with new kubecontext for the GKE cluster:
+
+   .. code-block:: console
+
+     $ gcloud container clusters get-credentials <cluster-name> --region <region>
+
+     # Example:
+     # gcloud container clusters get-credentials testcluster --region us-central1-c
+
+3. [If using GPUs] If your GKE nodes have GPUs, you may need to to
+   `manually install <https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/>`_
+   nvidia drivers. You can do so by deploying the daemonset
+   depending on the GPU and OS on your nodes:
+
+   .. code-block:: console
+
+     # For Container Optimized OS (COS) based nodes with GPUs other than Nvidia L4 (e.g., V100, A100, ...):
+     $ kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/nvidia-driver-installer/cos/daemonset-preloaded.yaml
+
+     # For Container Optimized OS (COS) based nodes with L4 GPUs:
+     $ kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/nvidia-driver-installer/cos/daemonset-preloaded-latest.yaml
+
+     # For Ubuntu based nodes with GPUs other than Nvidia L4 (e.g., V100, A100, ...):
+     $ kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/nvidia-driver-installer/ubuntu/daemonset-preloaded.yaml
+
+     # For Ubuntu based nodes with L4 GPUs:
+     $ kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/nvidia-driver-installer/ubuntu/daemonset-preloaded-R525.yaml
+
+   To verify if GPU drivers are set up, run ``kubectl describe nodes`` and verify that ``nvidia.com/gpu`` is listed under the ``Capacity`` section.
+
+4. Verify your kubeconfig (and GPU support, if available) is correctly set up by running :code:`sky check`:
+
+   .. code-block:: console
+
+     $ sky check
+
```

Comment:
5. Check available GPUs in the kubernetes cluster with `sky show-gpus --cloud kubernetes`